pkg/server/plugin/keymanager/gcpkms/gcpkms_test.go

```diff
@@ -0,0 +1,1478 @@
+								Algorithm: kmspb.CryptoKeyVersion_EC_SIGN_P256_SHA256,
+								Name:      fmt.Sprintf("%s/cryptoKeyVersions/1", cryptoKeyName2),
+							}},
+					},
+				},
+			},
+		},
+		{
+			name:              "list CryptoKeys error",
+			configureRequest:  configureRequestWithDefaults(t),
+			err:               "list keys failure",
+			listCryptoKeysErr: errors.New("list keys failure"),
+			fakeCryptoKeys: []fakeCryptoKey{
+				{
+					CryptoKey: &kmspb.CryptoKey{
+						Name:            cryptoKeyName1,
+						Labels:          make(map[string]string),
+						VersionTemplate: &kmspb.CryptoKeyVersionTemplate{Algorithm: kmspb.CryptoKeyVersion_EC_SIGN_P256_SHA256},
+					},
+					fakeCryptoKeyVersions: map[string]*fakeCryptoKeyVersion{
+						"1": {
+							publicKey: pubKey,
+							CryptoKeyVersion: &kmspb.CryptoKeyVersion{
+								Algorithm: kmspb.CryptoKeyVersion_EC_SIGN_P256_SHA256,
+								Name:      fmt.Sprintf("%s/cryptoKeyVersions/1", cryptoKeyName1),
+							}},
+					},
+				},
+			},
+		},
+	} {
+		tt := tt
+		t.Run(tt.name, func(t *testing.T) {
+			ts := setupTest(t)
+			ts.fakeKMSClient.putFakeCryptoKeys(tt.fakeCryptoKeys)
+
+			// This is so dispose aliases blocks on init and allows to test dispose keys isolated
+			ts.plugin.hooks.disposeCryptoKeysSignal = make(chan error)
+			disposeCryptoKeysSignal := make(chan error)
+			ts.plugin.hooks.disposeCryptoKeysSignal = disposeCryptoKeysSignal
+			scheduleDestroySignal := make(chan error)
+			ts.plugin.hooks.scheduleDestroySignal = scheduleDestroySignal
+
+			// exercise
+			_, err := ts.plugin.Configure(ctx, tt.configureRequest)
+			require.NoError(t, err)
+
+			ts.fakeKMSClient.listCryptoKeysErr = tt.listCryptoKeysErr
+
+			// wait for dispose keys task to be initialized
+			_ = waitForSignal(t, disposeCryptoKeysSignal)
+			// move the clock forward so the task is run
+			ts.clockHook.Add(48 * time.Hour)
+			// wait for dispose keys to be run
+			err = waitForSignal(t, disposeCryptoKeysSignal)
+			// assert errors
+			if tt.err != "" {
+				require.NotNil(t, err)
+				require.Equal(t, tt.err, err.Error())
+				return
+			}
+			// wait for schedule delete to be run
+			_ = waitForSignal(t, scheduleDestroySignal)
+
+			// assert
+			for _, cryptoKey := range ts.fakeKMSClient.store.fakeCryptoKeys {
+				require.Equal(t, cryptoKey.Labels[labelNameActive], "false")
+			}
+		})
+	}
+}
+
+func TestGenerateKey(t *testing.T) {
+	for _, tt := range []struct {
+		configureReq   *configv1.ConfigureRequest
+		expectCode     codes.Code
+		expectMsg      string
+		destroyTime    *timestamp.Timestamp
+		fakeCryptoKeys []fakeCryptoKey
+		generateKeyReq *keymanagerv1.GenerateKeyRequest
+		logs           []spiretest.LogEntry
+		name           string
+		waitForDelete  bool
+
+		createKeyErr               error
+		destroyCryptoKeyVersionErr error
+		getCryptoKeyVersionErr     error
+		getPublicKeyErr            error
+		getTokenInfoErr            error
+		updateCryptoKeyErr         error
+	}{
```

Comment:
I dont know if this is the place, but, can you add a test case to validate what happens if we are not able to enqueue items for destruction? and verify that cryptoKey is disabled "only" when it does not contain vesions?